align_system/utils/kde_utils.py

```diff
@@ -13,19 +13,47 @@
 KDE_BANDWIDTH = 0.75 * (KDE_MAX_VALUE / 10.0)
 
 
-def load_kde(target_kdma, norm='globalnorm'):
+def load_kde(target_kdma, norm='globalnorm', norm_factor=0.5):
     if isinstance(target_kdma, KDMAValue):
         target_kdma = target_kdma.to_dict()
 
     if norm == 'globalnorm':
-        target_kde_base64 = target_kdma['kdes']['globalnorm']['kde']
+        target_kde = kde_from_base64(target_kdma['kdes']['globalnorm']['kde'])
     elif norm == 'localnorm':
-        target_kde_base64 = target_kdma['kdes']['localnorm']['kde']
+        target_kde = kde_from_base64(target_kdma['kdes']['localnorm']['kde'])
     elif norm == 'rawscores':
-        target_kde_base64 = target_kdma['kdes']['rawscores']['kde']
+        target_kde = kde_from_base64(target_kdma['kdes']['rawscores']['kde'])
+    elif norm == 'priornorm':
+        # Load target KDE and get density
+        kde = kde_from_base64(target_kdma['kdes']['rawscores']['kde'])
+        linspace = np.linspace(0, 1, 1000)
+        density = _kde_to_pdf(kde, linspace)
+
+        # Get prior KDE and density
+        #TODO - get prior data counts from train yamls rather than precomputing and hard coding
+        if target_kdma['kdma'] == 'QualityOfLife':
+            prior_data = [0.1]*24 + [0.3]*12 + [0.7]*12 + [0.9]*24
+        elif target_kdma['kdma'] == 'PerceivedQuantityOfLivesSaved':
+            prior_data = [0.1]*24 + [0.3]*16 + [0.7]*8 + [0.9]*24
+        else:
+            raise RuntimeError(f'No prior data for {target_kde['kdma']}')
```

Comment:
This is a good error, so for example if we try to run prior norm on the ADEPT stuff it should error out loudly right?

Reply:
Yes running with ADEPT would trigger this error